src/Compilers/Core/Portable/DiagnosticAnalyzer/IAnalyzerPathResolver.cs

```diff
@@ -0,0 +1,56 @@
+﻿
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// This interface allows hosts to control where an analyzer is loaded from. It can redirect the path 
+    /// originally passed to the compiler to a new path. Or it can take ownership of a path to prevent 
```

Comment:
I'm not sure what the `or` here is representing. AFAICT, the way this interface works is first one to return true from `IsAnalyzerPathHandled` wins, and no other APIs on this interface will be called if that returned false.

Reply:
Consider the `ProgramFilesAnalyzerPathResolver` as the example of that 'or' block. It does absolutely nothing to the path, it just says that it handles the path. That prevents later handlers, like shadow copy, from ever seeing it.

Reply:
This reminds me of "[interceptors](https://square.github.io/okhttp/features/interceptors/)" in the OkHttp sense.

Reply:
> there isn't "composition" of the path handling 

I did consider that pattern but there were no such composition patterns that I could think of. Didn't want to add in extensibility that we weren't using. If we do find such a composition case it should be rather minor change to accomodate it. All of these are internal APIs

Reply:
> Think this needs to be clarified explicitly. In both of these cases, the resolver is taking ownership of the path.

The next paragraph is 

```csharp
    /// Instances of these types are considered in the order they are added to the <see cref="AnalyzerAssemblyLoader"/>.
    /// The first instance to return true from <see cref="IsAnalyzerPathHandled(string)"/> will be considered to 
    /// be the owner of that path. 
```

This says explictily that it's the owner of the path. What part isn't clarified hhere?

Reply:
I don't think either of us were suggesting actually doing such a thing; my problem is that the comment as-worded _suggests_ that we are doing that more complicated strategy.

Reply:
> What part isn't clarified hhere?

> It can redirect the path 
>    /// originally passed to the compiler to a new path. Or it can take ownership of a path to prevent

There's no `or` here. Both of these scenarios take ownership of the path. Whether it does a no-op or not is irrelevant to the interface.